### PR TITLE
Make Request Errors 500

### PR DESCRIPTION
**Pull request by hakimms**
This PR will update request validation errors to send back 500s to eSTS and response validation errors to send back 400s. This way, we can identify that 500 errors is eSTS are internal and should be marked as failures whereas 400 errors are customer input errors and should be identified as CallerErrors in eSTS.

[Bug 2161532](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2161532): Treat Only Response Validation Errors as 500

### CODE REVIEW

File: sdk/identity/Azure.Identity.BrokeredAuthentication/api/Azure.Identity.BrokeredAuthentication.net462.cs [hunk 1]
```diff
@@ -3,10 +3,12 @@ namespace Azure.Identity.BrokeredAuthentication
     public partial class InteractiveBrowserCredentialBrokerOptions : Azure.Identity.InteractiveBrowserCredentialOptions
     {
         public InteractiveBrowserCredentialBrokerOptions(System.IntPtr parentWindowHandle) { }
+        public bool? IsMsaPassthroughEnabled { get { throw null; } set { } }
```
Comment (HarmanDhunna) [inline sdk/identity/Azure.Identity.BrokeredAuthentication/api/Azure.Identity.BrokeredAuthentication.net462.cs:6]: Revert

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventMetadata.cs [hunk 2]
```diff
@@ -71,7 +72,14 @@ internal AuthenticationEventRequestBase CreateEventRequest(HttpRequestMessage re
 
             if (validate)
             {
-                Helpers.ValidateGraph(eventRequest);
+                try
+                {
+                    Helpers.ValidateGraph(eventRequest);
+                }
+                catch (ValidationException exception)
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventMetadata.cs:79]: Comments why we are doing this.

Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventMetadata.cs:79]: Why not throw `RequestValidationException` from `ValidateGraph`?

**Comment by HarmanDhunna** [comment]
Please update the description with more details. For example, why is it we want response validation to be 500?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs [hunk 3]
```diff
@@ -69,7 +68,7 @@ public Task SetValueAsync(object result, CancellationToken cancellationToken)
                     AuthenticationEventResponse response = Request.GetResponseObject();
                     if (response == null)
                     {
-                        throw new InvalidOperationException(AuthenticationEventResource.Ex_Missing_Request_Response);
+                        throw new RequestValidationException(AuthenticationEventResource.Ex_Missing_Request_Response);
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs:71]: for the catch, are there any other exception we will have other than ResponseValidationException?

**Review by HarmanDhunna — CHANGES_REQUESTED** [review]

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventMetadata.cs (hunk 2 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventMetadata.cs:79]: ValidateGraph is generic and validates other objects like the response as well.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs (hunk 3 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs:71]: We also throw UnauthorizedAccessException if the tokens are not valid

File: sdk/identity/Azure.Identity.BrokeredAuthentication/api/Azure.Identity.BrokeredAuthentication.net462.cs (hunk 1 above)
Comment (hakimms) [reply sdk/identity/Azure.Identity.BrokeredAuthentication/api/Azure.Identity.BrokeredAuthentication.net462.cs]: Done

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs (hunk 3 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs:71]: We also throw an UnauthorizedAccessException if the tokens are invalid

**Comment by HarmanDhunna** [comment]
Love the description. Thank you. What are you thoughts we make it a pattern of storing this kind of information in the ChangeLog.md as well. So when we go back and write documentation or how-to-guide, it's all in one place?

**Comment by hakimms** [comment]
> Love the description. Thank you. What are you thoughts we make it a pattern of storing this kind of information in the ChangeLog.md as well. So when we go back and write documentation or how-to-guide, it's all in one place?

I like the idea. I will update the changelog file. Let me see how other azure sdk instances have written their changelog files

**Pull request merged**